src/util.c

```diff
@@ -163,14 +162,11 @@ void sleep(uint16_t t)
     ; loop for sleep
     ; loop from (0xFFFF - t) to (0xFFFF)
     LOOP$:
-        lcall __sleep_1ms               ;#8*(frep/12000) - 10
+        lcall __sleep_1ms               ;#5*(ar7*(256 + 1) + ar6)
         inc dptr                        ;#2
         mov a,dpl                       ;#1
         anl a,dph                       ;#1
         cpl a                           ;#1
```

Comment:
Linear growth can be achieved after adding a  nop, could you possibly modify it?